source/Sale.h

```diff
@@ -74,4 +76,12 @@ bool Sale<Item>::Has(const Item *item) const
 
 
 
+template <class Item>
+const Item *Sale<Item>::Sample() const
+{
+	auto it = this->begin();
+	std::advance(it, Random::Int(this->size()));
+	return (it == this->end() ? nullptr : *it);
```

Comment:
`it` will never be the end iterator, because the index will always be to a valid element. You can also simplify it like this:

```suggestion
	return std::next(this->begin(), Random::Int(this->size()));
```

Same goes for the function in `Set`.

Reply:
Just realized it can be the end iterator if the Sale or Set is empty. I think the original version is needed.

Reply:
oh whoops,, you're right sorry. mb

Reply:
Will switch up with a guard of size zero and then your statement. Its easier to understand.

Reply:
I think the final version here is more readable then how it started out.